R/tbl-df.r

```diff
@@ -2,6 +2,44 @@
 #' @exportClass tbl_df
 setOldClass(c("tbl_df", "tbl", "data.frame"))
 
+#' `tbl_df` class
+#'
+#' @description
+#' Objects of class `tbl_df` have a class attribute consisting of `tbl_df`,
+#' `tbl`, and `data.frame`. So-called tibbles are a special flavor of
+#' [`data.frame`][base::data.frame()], created in order to have different
+#' default behaviour, detailed below.
+#'
+#' @section tibbles vs. `data.frame`s:
+#'
+#' How default behaviour of tibbles differs from that of data frames, during
+#' creation, access, and modification:
+#'
+#' * Column data is not coerced. A character vector is not turned into a factor.
+#' * Column names are not munged, although missing names are auto-populated.
+#' Duplicated column names are strongly discouraged. Read about [name-repair].
```

Comment:
I haven't done this because the current formatting is what I get when I Reflow Comment in RStudio. Do you really want us to go "manual" on this? I agree that would look better, though!

Reply:
```
*  First line
   Second line
```

OK I've done this manually in all the bullet lists.